testing/time.ts

```diff
@@ -131,12 +107,12 @@ function fakeSetTimeout(
   // deno-lint-ignore no-explicit-any
   ...args: any[]
 ): number {
-  if (!time) throw new TimeError("Time is not faked");
+  if (!time) throw new ReferenceError("Time is not faked");
```

Comment:
Note: made to be similar to when a non-existent variable is referenced ([ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Not_defined)).

Reply:
I don't see the similarity with that situation. How about `Error`?

Reply:
Similarly to the case of restore, `time` variable referenced here is implementation detail, and not visible to the users.

Reply:
Fair point. Ok, what if we used a [`NotSupportedError`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException#notsupportederror) `DOMException` instead for these cases ("Time is not faked", "Time is already faked" and "Time is already restored")? Alternatively, we could keep `TimeError` for these cases, as standard JS errors don't seem to cover this use case. WDYT?

Reply:
`TimeError` sounds reasonable to me